src/main.rs

```diff
@@ -345,6 +347,43 @@ unsafe extern "C" fn load(cx: *mut JSContext, argc: u32, vp: *mut Value) -> bool
     return true;
 }
 
+unsafe extern "C" fn read(cx: *mut JSContext, argc: u32, vp: *mut Value) -> bool {
+    if argc < 1 {
+        return false;
+    }
+
+    let args = CallArgs::from_vp(vp, argc);
+    let val = args.get(0);
+    let s = js::rust::ToString(cx, val);
+    if s.is_null() {
+        // TODO: report error
+        return false;
+    }
+
+    let mut filename = env::current_dir().unwrap();
+    let path_root = Rooted::new(cx, s);
+    let path = JS_EncodeStringToUTF8(cx, path_root.handle());
+    let path = CStr::from_ptr(path);
+    filename.push(str::from_utf8(path.to_bytes()).unwrap());
```

Comment:
Would be good to factor this out: it's an exact duplicate of the same 5 lines in `load` above. That can well be a follow-up, though.